src/Events.Functions/Clients/EventsStorageClient.cs

```diff
@@ -0,0 +1,80 @@
+using Altinn.Platform.Events.Functions.Configuration;
+using Altinn.Platform.Events.Functions.Extensions;
+using Altinn.Platform.Events.Functions.Models;
+using Altinn.Platform.Events.Functions.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.Platform.Events.Functions.Clients
+{
+    /// <summary>
+    /// Events Storage REST API client
+    /// </summary>
+    public class EventsStorageClient : IEventsStorageClient
+    {
+        private readonly HttpClient _client;
+        private readonly IAccessTokenGenerator _accessTokenGenerator;
+        private readonly IKeyVaultService _keyVaultService;
+        private readonly KeyVaultSettings _keyVaultSettings;
+        private readonly PlatformSettings _platformSettings;
+        private readonly ILogger<IPushToOutboundQueueClient> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PushToOutboundQueueClient"/> class.
+        /// </summary>
+        public EventsStorageClient(
+            HttpClient httpClient,
+            IAccessTokenGenerator accessTokenGenerator,
+            IKeyVaultService keyVaultService,
+            IOptions<PlatformSettings> eventsConfig,
+            IOptions<KeyVaultSettings> keyVaultSettings,
+            ILogger<IPushToOutboundQueueClient> logger)
+        {
+            _platformSettings = eventsConfig.Value;
+            _keyVaultSettings = keyVaultSettings.Value;
+            httpClient.BaseAddress = new Uri(_platformSettings.ApiEventsEndpoint);
+            _client = httpClient;
+            _accessTokenGenerator = accessTokenGenerator;
+            _keyVaultService = keyVaultService;
+            _logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public async Task SaveCloudEvent(CloudEvent item)
+        {
+            StringContent httpContent = new(JsonSerializer.Serialize(item), Encoding.UTF8, "application/json");
+            try
+            {
+                string endpointUrl = "storage/events";
+
+                string certBase64 = await _keyVaultService.GetCertificateAsync(_keyVaultSettings.KeyVaultURI, _keyVaultSettings.PlatformCertSecretId);
+                string accessToken = _accessTokenGenerator.GenerateAccessToken(
+                    "platform",
+                    "events",
+                    new X509Certificate2(Convert.FromBase64String(certBase64), (string)null, X509KeyStorageFlags.MachineKeySet | X509KeyStorageFlags.PersistKeySet | X509KeyStorageFlags.Exportable));
+
+                HttpResponseMessage response = await _client.PostAsync(endpointUrl, httpContent, accessToken);
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    var msg = $"// SaveCloudEvent with id {item.Id} failed with statuscode {response.StatusCode}";
+                    _logger.LogError(msg);
+                    throw new Exception(msg);
```

Comment:
What's we've done for other clients communicating with platform is that we've created a custom exception type PlatformHttpException (I think) and then relayed the status code from the platform component. That could be done here as well. If not the HttpRequestException seems like the better option

Reply:
My suggestion was to create a seperate exceptin class in this project instead of creating a dependency between the project, but this HttpRequestException is good enough for now.